crates/apollo-mcp-server/src/graphql.rs

```diff
@@ -74,11 +79,21 @@ pub trait Executable {
             }
         }
 
-        reqwest::Client::new()
+        let client = ClientBuilder::new(reqwest::Client::new())
+            .with_init(Extension(OtelName("graphql-client".into())))
+            // Insert the tracing middleware
+            .with(TracingMiddleware::default())
+            .build();
+
+        client
             .post(request.endpoint.as_str())
+            .with_extension(OtelName(
+                format!("POST {}", request.endpoint.as_str()).into(),
+            ))
             .headers(self.headers(&request.headers))
             .body(Value::Object(request_body).to_string())
             .send()
+            .with_current_context()
```

Comment:
I'm curious about the order here. Is this called after `send()` so that both the request and response can be captured?